packages/compiler/core/types.ts

```diff
@@ -474,12 +510,14 @@ export const enum SymbolFlags {
   Using                 = 1 << 18,
   DuplicateUsing        = 1 << 19,
   SourceFile            = 1 << 20,
-
+  Declaration           = 1 << 21,
```

Comment:
This flag is confusing since I believe all symbols hold declarations. Can the assumption be its declaration unless the implementation flag is present?

Reply:
well the idea here is that the `extern dec` sets `Declaration` and the js symbol sets `implementation` so you need to have both (or just implementation) to have a valid decorator but shouldn;'t merge 2 declaration or 2 implementation together.

Reply:
I understand the idea (I think, anyway), but why does that mean we need to track declaration as a separate flag? I believe all symbols are declarations.

Reply:
yeah I guess everything would be a declaration, technically even a JS function is declaring itself just with no info.

would maybe then `Signature` make more sense maybe?

Reply:
Signature definitely makes more sense, but I guess just make sure you need both flags. Symbol flags are limited real estate!

Reply:
that's fair, I think I need both if we want to do the check using SymbolsFlags. Could also check the list of declarations and make sure that the is only 1 declaration node and 1 implementation node otherwise emit duplicate.